### PR TITLE
Add VS Code settings for configuration properties

### PR DESCRIPTION
**Pull request by bobbrow**
This adds the new settings for c_cpp_properties.json to implement the spec in #1338 

(I haven't been able to write any tests for this because the vscode.WorkspaceConfiguration API doesn't persist my attempts to update settings.  I have reached out to the VS Code team for help.)

### CODE REVIEW

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -177,6 +177,9 @@ function collectSettingsForTelemetry(filter: (key: string, val: string, settings
                     break;
                 }
                 default: {
+                    if (key.startsWith("default.")) {
+                        continue;   // Don't log c_cpp_properties.json defaults since they may contain PII.
```
Comment (bobbrow) [inline Extension/src/LanguageServer/client.ts:181]: Open Issue: Do we want to log a generic event to track feature usage though?

File: Extension/package.json [hunk 2]
```diff
@@ -229,6 +229,118 @@
...
+          "description": "The value to use in a configuration if \"browse.databaseFilename\" is not specified, or the values to insert if \"${default}\" is present in \"browse.databaseFilename\".",
+          "scope": "resource"
+        },
+        "C_Cpp.default.browse.limitSymbolsToIncludedHeaders": {
+          "type": "boolean",
+          "default": true,
+          "description": "The value to use in a configuration if \"browse.limitSymbolsToIncludedHeaders\" is not specified, or the values to insert if \"${default}\" is present in \"browse.limitSymbolsToIncludedHeaders\".",
+          "scope": "resource"
+        },
+        "C_Cpp.default.systemIncludePath": {
```
Comment (sean-mcmanus) [inline Extension/package.json:369]: Why is this needed again? I keep forgetting. Oh, is this for cl.exe/Windows paths?

File: Extension/src/LanguageServer/client.ts (hunk 1 above)
Comment (sean-mcmanus) [reply Extension/src/LanguageServer/client.ts:181]: I don't know...up to your or Rong.

**Review by sean-mcmanus — APPROVED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Comment by sean-mcmanus** [comment]
Should we publish newer binaries so people running the extension from the typescript aren't broken? Maybe we can just wait till people complain...

**Comment by sean-mcmanus** [comment]
Oh -- can you add this back to activationEvents: `"workspaceContains:**/c_cpp_properties.json"`, adding `"workspaceContains:**/compile_commands.json"` also seems good (assuming that doesn't exist in non-C/C++ projects)...`"workspaceContains:**/cpp.hint"` would also be valid but lower pri...we don't want `"workspaceContains:**/*.cpp"` due to the scenario where a primarily non-C/C++ project happens to have some subfolder with C/C++.
